code/modules/antagonists/changeling/powers/revive.dm

```diff
@@ -11,6 +11,10 @@
 	if(HAS_TRAIT(user, TRAIT_UNREVIVABLE))
 		to_chat(user, "<span class='notice'>Something is preventing us from regenerating, we will need to revive at another point.</span>")
 		return FALSE
+
+	cling.can_use_powers = FALSE
+	addtimer(VARSET_CALLBACK(cling, can_use_powers, TRUE), 10 SECONDS)
```

Comment:
You could probably just store the timestamp of the last time the changeling has revived
then check `world.time > timestamp + 10 SECONDS` to see if 10s have passed since then